### PR TITLE
uses promises instead of callbacks

### PR DESCRIPTION
**Pull request by gonber**
BREAKING CHANGE

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,12 +1,13 @@
 {
   "name": "@funnode/slack",
-  "version": "1.5.0",
+  "version": "2.0.0",
```
Comment (RaviPatel) [inline package.json:3]: let's avoid updating the version until after the PR is approved and is being merged

File: .npmignore [hunk 2]
```diff
@@ -0,0 +1 @@
+test/
```
Comment (RaviPatel) [inline .npmignore:1]: code coverage tests are failing I think because it can't find these tests - are you leaving them out for another PR?

Comment (gonber) [reply .npmignore:1]: yes, they're ready to push whenever the next milestone opens on upwork. I've got to 100% coverage

File: package.json (hunk 1 above)
Comment (gonber) [reply package.json:3]: sure, I'll follow that for the other repos

Comment (RaviPatel) [reply package.json:3]: Sounds good .. but I'll have to either leave this PR open until the tests update or we'll have to re-add the tests directory in order for the github actions to pass

Comment (gonber) [reply package.json:3]: I'm confused now. This PR does not remove the tests directory. It leaves it as it is

Comment (RaviPatel) [reply package.json:3]: Hmm, I thought the failure was related to these changes: https://github.com/FunNode/npm-slack/pull/3/checks?check_run_id=337801422. If not, don't worry about it

Comment (gonber) [reply package.json:3]: the culprit seems to be
> ./cc-reporter after-build --exit-code 0
Error: you must supply a CC_TEST_REPORTER_ID ENV variable or pass it via the -r flag

Comment (RaviPatel) [reply package.json:3]: Oh, got it - the action is running on your #master which likely doesn't have the ENV set. It'll work after merging to FunNode master, so we're good - thanks

**Review by RaviPatel — APPROVED** [review]

**Pull request merged**